From 5fdccf13bf69b9934aed782c6e6b2644410de062 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Reto=20Schu=CC=88ttel?= <reto@schuettel.ch>
Date: Wed, 4 Nov 2009 14:07:30 +0100
Subject: [PATCH 2/2] Use correct quote_value method

Use the @owner's quote_value method when a :primary_key has been
specified in an :has_many-through association.
---
 .../associations/has_many_association.rb           |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/associations/has_many_association.rb b/activerecord/lib/active_record/associations/has_many_association.rb
index 1f673ad..db2aa2b 100644
--- a/activerecord/lib/active_record/associations/has_many_association.rb
+++ b/activerecord/lib/active_record/associations/has_many_association.rb
@@ -8,7 +8,7 @@ module ActiveRecord
       protected
         def owner_quoted_id
           if @reflection.options[:primary_key]
-            quote_value(@owner.send(@reflection.options[:primary_key]))
+            @owner.class.quote_value(@owner.send(@reflection.options[:primary_key]))
           else
             @owner.quoted_id
           end
-- 
1.6.5.1

